library/network_state.py

```diff
@@ -0,0 +1,74 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+# SPDX-License-Identifier: BSD-3-Clause
+
+from __future__ import absolute_import, division, print_function
+
+__metaclass__ = type
+
+DOCUMENTATION = r"""
+---
+module: network_state
+version_added: "2.9"
+short_description: module for network role to apply network state configuration
+description:
+    - This module allows to apply the network state configuration through nmstate,
+      https://github.com/nmstate/nmstate
+options:
+    state:
+        description: Nmstate state definition
+        required: true
+        type: dict
+author: "Wen Liang (@liangwen12year)"
+"""
+
+RETURN = r"""
+state:
+    description: Network state after running the module
+    type: dict
+    returned: always
+"""
+
+from ansible.module_utils.basic import AnsibleModule
+import libnmstate  # pylint: disable=import-error
+
+
+class NetworkState:
+    def __init__(self, module, module_name):
+        self.module = module
+        self.params = module.params
+        self.result = dict(changed=False)
+        self.module_name = module_name
+        self.previous_state = libnmstate.show()
+
+    def run(self):
+        state = self.params["state"]
+        libnmstate.apply(state)
+        current_state = libnmstate.show()
+        if current_state != self.previous_state:
+            self.result["changed"] = True
+
+        self.result["state"] = current_state
+
+        self.module.exit_json(**self.result)
+
+
+def run_module():
+    module_args = dict(
+        state=dict(type="dict", required=True),
```

Comment:
I think a better name would be `desired_state` because is what the user is specifying. `state` seems a little bit ambiguous..

Reply:
I changed into `desired_state`.